akka-actor/src/main/scala/akka/io/dns/internal/TcpDnsClient.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns.internal
+
+import java.net.InetSocketAddress
+
+import akka.AkkaException
+import akka.actor.{ Actor, ActorLogging, ActorRef, Stash }
+import akka.annotation.InternalApi
+import akka.io.Tcp._
+import akka.io.dns.internal.DnsClient.{ Answer, DnsQuestion, Question4 }
+import akka.io.{ IO, Tcp }
+import akka.util.ByteString
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class TcpDnsClient(tcp: ActorRef, ns: InetSocketAddress, answerRecipient: ActorRef) extends Actor with ActorLogging with Stash {
+  import TcpDnsClient._
+
+  import context.system
+
+  override def receive: Receive = idle
+
+  val idle: Receive = {
+    case _: Message ⇒
+      stash()
```

Comment:
I guess the stash will never get too big if the dns server doesn't support TCP as we'll restart back a backoff